tests/sqlite3/test_wal.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from dissect.database.sqlite3 import sqlite3
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+
+def test_sqlite_wal_binaryio(sqlite_db: Path, sqlite_wal: Path) -> None:
+    s = sqlite3.SQLite3(sqlite_db.open("rb"), sqlite_wal.open("rb"), checkpoint=1)
+    _sqlite_read_checkpoint_1(s)
+
+    s = sqlite3.SQLite3(sqlite_db.open("rb"), sqlite_wal.open("rb"), checkpoint=2)
+    _sqlite_read_checkpoint_2(s)
+
+    s = sqlite3.SQLite3(sqlite_db.open("rb"), sqlite_wal.open("rb"), checkpoint=3)
+    _sqlite_read_checkpoint_3(s)
+
+
+def test_sqlite_wal_auto_detect_binaryio(sqlite_db: Path) -> None:
+    s = sqlite3.SQLite3(sqlite_db.open("rb"), checkpoint=1)
+    _sqlite_read_checkpoint_1(s)
+
+    s = sqlite3.SQLite3(sqlite_db.open("rb"), checkpoint=2)
+    _sqlite_read_checkpoint_2(s)
+
+    s = sqlite3.SQLite3(sqlite_db.open("rb"), checkpoint=3)
+    _sqlite_read_checkpoint_3(s)
+
+
+def test_sqlite_wal_path(sqlite_db: Path, sqlite_wal: Path) -> None:
+    s = sqlite3.SQLite3(sqlite_db, sqlite_wal, checkpoint=1)
+    _sqlite_read_checkpoint_1(s)
+
+    s = sqlite3.SQLite3(sqlite_db, sqlite_wal, checkpoint=2)
+    _sqlite_read_checkpoint_2(s)
+
+    s = sqlite3.SQLite3(sqlite_db, sqlite_wal, checkpoint=3)
+    _sqlite_read_checkpoint_3(s)
+
+
+def test_sqlite_wal_auto_detect_path(sqlite_db: Path) -> None:
+    s = sqlite3.SQLite3(sqlite_db, checkpoint=1)
+    _sqlite_read_checkpoint_1(s)
+
+    s = sqlite3.SQLite3(sqlite_db, checkpoint=2)
+    _sqlite_read_checkpoint_2(s)
+
+    s = sqlite3.SQLite3(sqlite_db, checkpoint=3)
+    _sqlite_read_checkpoint_3(s)
+
+
+def _sqlite_read_checkpoint_1(s: sqlite3.SQLite3) -> None:
```

Comment:
```suggestion
def _assert_checkpoint_1(s: sqlite3.SQLite3) -> None:
```
Make the intention a bit clearer.